test/PrintAsObjC/protocols.swift

```diff
@@ -187,6 +187,20 @@ extension NSString : A, ZZZ {}
 // CHECK: @interface RootClass4 <A, ZZZ>{{$}}
 @objc class RootClass4 : A, ZZZ {}
 
+// CHECK-LABEL: @interface RootClass4 (SWIFT_EXTENSION(protocols))
+// CHECK-NEXT: @end
+extension RootClass4 {
+
+  // CHECK-LABEL: @protocol DefinedInExtension
+  // CHECK-NEXT: @end
+  @objc protocol DefinedInExtension {}
+
+  // CHECK-LABEL: SWIFT_PROTOCOL_NAMED("DefinedInExtension2")
```

Comment:
I have a feeling this should be `RootClass4.DefinedInExtension2`, but I'm not certain about how the `swift_name` attribute is used. In any case the same happens for classes:

```swift
class Outer: NSObject { @objc(CustomClassName) class Inner: NSObject {} }
```
Generates the header entry:
```objc
SWIFT_CLASS_NAMED("Inner") // <-- Not "Outer.Inner"
@interface CustomClassName : NSObject
- (nonnull instancetype)init OBJC_DESIGNATED_INITIALIZER;
@end
```